src/helpers/__tests__/accessiblity.test.tsx

```diff
@@ -1,175 +1,237 @@
+    const view = render(
       <View>
+        <View accessibilityViewIsModal />
         <View>
-          <View testID="subject" />
+          <View>
+            <View testID="subject" />
+          </View>
         </View>
       </View>
-    </View>
-  );
-  expect(isHiddenFromAccessibility(view.getByTestId('subject'))).toBe(true);
-});
+    );
+    expect(isHiddenFromAccessibility(view.getByTestId('subject'))).toBe(true);
+  });
 
-test('is not triggered for element with accessibilityViewIsModal prop', () => {
-  const view = render(
-    <View>
-      <View accessibilityViewIsModal testID="subject" />
-    </View>
-  );
-  expect(isHiddenFromAccessibility(view.getByTestId('subject'))).toBe(false);
-});
+  test('is not triggered for element with accessibilityViewIsModal prop', () => {
+    const view = render(
+      <View>
+        <View accessibilityViewIsModal testID="subject" />
+      </View>
+    );
+    expect(isHiddenFromAccessibility(view.getByTestId('subject'))).toBe(false);
+  });
 
-test('is not triggered for child of element with accessibilityViewIsModal prop', () => {
-  const view = render(
-    <View>
-      <View accessibilityViewIsModal>
-        <View testID="subject" />
+  test('is not triggered for child of element with accessibilityViewIsModal prop', () => {
+    const view = render(
+      <View>
+        <View accessibilityViewIsModal>
+          <View testID="subject" />
+        </View>
       </View>
-    </View>
-  );
-  expect(isHiddenFromAccessibility(view.getByTestId('subject'))).toBe(false);
-});
+    );
+    expect(isHiddenFromAccessibility(view.getByTestId('subject'))).toBe(false);
+  });
 
-test('is not triggered for descendent of element with accessibilityViewIsModal prop', () => {
-  const view = render(
-    <View>
-      <View accessibilityViewIsModal>
-        <View>
+  test('is not triggered for descendent of element with accessibilityViewIsModal prop', () => {
+    const view = render(
+      <View>
+        <View accessibilityViewIsModal>
           <View>
-            <View testID="subject" />
+            <View>
+              <View testID="subject" />
+            </View>
           </View>
         </View>
       </View>
-    </View>
-  );
-  expect(isHiddenFromAccessibility(view.getByTestId('subject'))).toBe(false);
-});
-
-test('has isInaccessible alias', () => {
-  expect(isInaccessible).toBe(isHiddenFromAccessibility);
+    );
+    expect(isHiddenFromAccessibility(view.getByTestId('subject'))).toBe(false);
+  });
+
+  test('has isInaccessible alias', () => {
+    expect(isInaccessible).toBe(isHiddenFromAccessibility);
+  });
+});
+
+describe('isAccessibilityElement', () => {
+  describe('when accessible prop is not defined', () => {
+    test('returns true for Text component', () => {
+      const element = render(<Text testID="text">Hey</Text>).getByTestId(
+        'text'
+      );
+      expect(isAccessibilityElement(element)).toEqual(true);
+    });
```

Comment:
How about structuring the tests a bit different form, by grouping default/true/false tests for given component type together. These tests will be both exhaustive and take less LOCs.

```suggestion
  test('TextInput matching', () => {
    const { getByTestId } = render(<View>
      <Text testID="default">Default</Text>
      <Text testID="true">True</Text>
      <Text testID="false">False</Text>
    </View>);
    expect(isAccessibilityElement(getByTestId("default"))).toBeTrue();
    expect(isAccessibilityElement(getByTestId("true"))).toBeTrue();
    expect(isAccessibilityElement(getByTestId("false"))).toBeFalse();
  });
```

Reply:
We could have such tests for key views that render to host views: View, Text, TextInput, Switch.

Maybe we can also add tests for composite components that render to View with accessible prop: Pressable, TouchableOpacity.

Reply:
good idea! the only downside of this method is that tests title are less explicit

Reply:
Now : ![CleanShot 2023-01-17 at 11 57 10](https://user-images.githubusercontent.com/32499425/212881571-1b171846-af3e-4874-8560-284ec2df20e6.png)
Before : 
![CleanShot 2023-01-17 at 11 58 26](https://user-images.githubusercontent.com/32499425/212881811-528fd755-2dc8-4eb7-9dc0-903c359f2085.png)

Reply:
what do you prefer? both are fine for me